lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -1361,31 +1361,19 @@ namespace Js
 
         Assert(!(callInfo.Flags & CallFlags_New));
 
-        //
-        // TODO: Move argument processing into DirectCall with proper handling.
-        //
+        PCWSTR const varName = L"String.prototype.match";
 
-        JavascriptString * pThis = nullptr;
-        GetThisStringArgument(args, scriptContext, L"String.prototype.match", &pThis);
+        AUTO_TAG_NATIVE_LIBRARY_ENTRY(function, callInfo, varName);
```

Comment:
For this function to show up in the debugger call stack. Most String functions don't call user-defined functions, so they don't need to be tagged. `String.prototype.replace`, for example, accepts a callback argument, so it's tagged. Since this function calls `RegExp.prototype[@@match]`, which can be overwritten by the user, it wouldn't show up in the call stack without the tag.